data-raw/simulate-data.R

```diff
@@ -411,27 +411,39 @@ simulate_data <- function(data, n) {
     list_cbind()
 }
 
-set.seed(123)
-simulation_definitions <- here("data-raw/simulation-definitions.csv") |>
-  read_csv(show_col_types = FALSE) |>
-  select(register_abbrev, variable_name, generator)
-
-simulation_definitions_list <- simulation_definitions |>
-  group_split(register_abbrev)
+#' Read simulation definitions from csv
+#'
+#' @param path Path to the csv file.
+#'
+#' @returns A tibble with simulation definitions.
+read_simulation_definitions_data <- function(path) {
+  path |>
+    readr::read_csv(show_col_types = FALSE) |>
+    dplyr::select("register_abbrev", "variable_name", "generator")
+}
 
-register_data <- simulation_definitions_list |>
-  map(\(data) simulate_data(data, n = 1000)) |>
-  map(insert_specific_atc) |>
-  map(add_fake_drug_name) |>
-  map(insert_false_drug_names) |>
-  map(insert_false_metformin) |>
-  map(insert_analysiscode) |>
-  # add the register abbreviations as a name to the list
-  set_names(
-    map_chr(
-      simulation_definitions_list,
-      \(data) unique(data$register_abbrev)
+set.seed(123)
+#' Create simulated register data
+#'
+#' @param simulation_definitions Tibble with simulation definitions.
+#'
+#' @returns A list with simulated register data.
+create_simulated_data <- function(simulation_definitions) {
+  simulation_definitions_list <- simulation_definitions |>
+    dplyr::group_split(register_abbrev)
+
+  register_data <- simulation_definitions_list |>
+    map(\(data) simulate_data(data, n = 1000)) |>
+    map(insert_specific_atc) |>
+    map(add_fake_drug_name) |>
+    map(insert_false_drug_names) |>
+    map(insert_false_metformin) |>
+    map(insert_analysiscode) |>
+    # add the register abbreviations as a name to the list
+    set_names(
+      map_chr(
+        simulation_definitions_list,
+        \(data) unique(data$register_abbrev)
+      )
     )
```

Comment:
Instead of two functions, this could be one (`create_simulated_data()`).

Reply:
You mean having the functionality of `read_simulated_data()` inside `create_simulated_data()`? 
I did it in two separate functions to mirror the functions for reading the variable descriptions and algorithm.

Reply:
I'll merge the functions into one 👍